src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,380 @@
+          .toArray
+        id -> newSerializedResponse
+      }.toMap
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.foreach {
+        case (id, peerApi) =>
+          peerApi.client
+            .getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF idToSerializedResponse(
+            id
+          )
+      }
+      val updateProps = redownloadService.fetchAndSetPeerProposals()
+      val newMajority = redownloadService.recalculateMajoritySnapshot()
+      val res = (updateProps >> newMajority).unsafeRunSync()
+
+      val correctSnaps = newMajorityProposals.sortBy { case (id, snap) => (-snap.height, snap.hash) }.map(_._2).head
+      val correctPeer = newMajorityProposals.sortBy { case (id, snap)  => (-snap.height, snap.hash) }.map(_._1).head
+      res.sortedSnaps.minBy { case snap => -snap.height } shouldBe correctSnaps
+      res.nodeIdsWithSnaps shouldBe peersWithnNewSnap
+    }
+
+    "should return correct majority snapshot when encountering distinct proposals at a given height" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      val newSetOfPeerSnaps = facilitators.zipWithIndex.map {
+        case ((id, peerInfo), idx) => (id, RecentSnapshot(s"$idx", 8L, Map.empty))
+      }
+      val idToSerializedResponse = facilitators.keys.map { id =>
+        val newProposal = baseSnapshots + (8L -> newSetOfPeerSnaps(id))
+        val newSerializedResponse = newProposal
+          .grouped(KryoSerializer.chunkSize)
+          .map(t => chunkSerialize(t.toSeq, RedownloadService.fetchSnapshotProposals))
+          .toArray
+        id -> newSerializedResponse
+      }.toMap
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.foreach {
+        case (id, peerApi) =>
+          (peerApi.client
+            .getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF serializedResponse)
+            .andThen(idToSerializedResponse(id))
+      }
+      val newProps = proposals ++ newSetOfPeerSnaps.toSeq
+      val updateProposals = redownloadService.fetchAndSetPeerProposals()
+      val oldMajority = redownloadService.recalculateMajoritySnapshot()
+      val updateProposalsAgain = redownloadService.fetchAndSetPeerProposals()
+      val newMajority = redownloadService.recalculateMajoritySnapshot()
+      val res = (updateProposals >> oldMajority >> updateProposalsAgain >> newMajority).unsafeRunSync()
+
+      val correctNewSnaps = newProps.sortBy { case (id, snap) => (-snap.height, snap.hash) }.map(_._2).head
+      val correctNewPeer = newProps.sortBy { case (id, snap)  => (-snap.height, snap.hash) }.map(_._1).head
+      res.sortedSnaps.minBy { case snap => -snap.height } shouldBe correctNewSnaps
+      res.nodeIdsWithSnaps shouldBe Set(correctNewPeer)
+    }
+
+    "should return empty diff if not enough snaps for a majority" in {
```

Comment:
Shouldn't `majority snapshot chooser` return a majority snapshot, but `check for alignment` checks for the snapshot diff?